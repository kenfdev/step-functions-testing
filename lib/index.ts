import { Construct } from 'constructs';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export interface StepFunctionsTestingProps {
  // Define construct properties here
}

export class StepFunctionsTesting extends Construct {

  constructor(scope: Construct, id: string, props: StepFunctionsTestingProps = {}) {
    super(scope, id);

    // Define construct contents here

    // example resource
    // const queue = new sqs.Queue(this, 'StepFunctionsTestingQueue', {
    //   visibilityTimeout: cdk.Duration.seconds(300)
    // });
  }
}
