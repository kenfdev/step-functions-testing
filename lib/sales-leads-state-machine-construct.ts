import { Construct } from 'constructs';
import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import { JsonPath, TaskInput } from 'aws-cdk-lib/aws-stepfunctions';
import * as ddb from 'aws-cdk-lib/aws-dynamodb';
import { Duration } from 'aws-cdk-lib';
import { DynamoAttributeValue } from 'aws-cdk-lib/aws-stepfunctions-tasks';

type StateMachineProps = {
  stateMachineName?: string;
  validationStateName?: string;
  checkIdentityStateName?: string;
  checkAddressStateName?: string;
  customValidationFailedStateName?: string;
  validationExceptionStateName?: string;
  detectSentimentStateName?: string;
  isPositiveSentimentStateName?: string;
  negativeSentimentDetectedStateName?: string;
  addToFollowUpStateName?: string;
  customerAddedToFollowupStateName?: string;
  retryInterval?: Duration;
};

export class SalesLeadsStateMachineConstruct extends Construct {
  constructor(scope: Construct, id: string, props: StateMachineProps = {}) {
    super(scope, id);

    const {
      validationStateName = 'Validation',
      checkIdentityStateName = 'Check Identity',
      checkAddressStateName = 'Check Address',
      customValidationFailedStateName = 'CustomValidationFailed',
      validationExceptionStateName = 'ValidationException',
      detectSentimentStateName = 'DetectSentiment',
      isPositiveSentimentStateName = 'Is Positive Sentiment?',
      negativeSentimentDetectedStateName = 'NegativeSentimentDetected',
      addToFollowUpStateName = 'Add to FollowUp',
      customerAddedToFollowupStateName = 'CustomerAddedToFollowup',
      stateMachineName = 'SalesLeadStateMachine',
      retryInterval = Duration.seconds(1),
    } = props;

    const checkIdentityState = new tasks.LambdaInvoke(
      this,
      checkIdentityStateName,
      {
        lambdaFunction: lambda.Function.fromFunctionArn(
          this,
          'CheckIdentityFunction',
          'arn:aws:lambda:us-west-2:123456789012:function:check-identity-function'
        ),
        inputPath: JsonPath.stringAt('$.data.identity'),
        payload: TaskInput.fromJsonPathAt('$'),
        resultSelector: {
          identity: JsonPath.stringToJson(JsonPath.stringAt('$.Payload.body')),
        },
        retryOnServiceExceptions: false,
      }
    ).addRetry({
      errors: [
        'Lambda.ServiceException',
        'Lambda.AWSLambdaException',
        'Lambda.SdkClientException',
        'CustomValidationError',
      ],
      backoffRate: 1,
      maxAttempts: 3,
      interval: retryInterval,
    });

    const checkAddressState = new tasks.LambdaInvoke(
      this,
      checkAddressStateName,
      {
        lambdaFunction: lambda.Function.fromFunctionArn(
          this,
          'CheckAddressFunction',
          'arn:aws:lambda:us-west-2:123456789012:function:check-address-function'
        ),
        inputPath: '$.data.address',
        payload: TaskInput.fromJsonPathAt('$'),
        resultSelector: {
          address: JsonPath.stringToJson(JsonPath.stringAt('$.Payload.body')),
        },
        retryOnServiceExceptions: false,
      }
    ).addRetry({
      errors: [
        'Lambda.ServiceException',
        'Lambda.AWSLambdaException',
        'Lambda.SdkClientException',
        'CustomValidationError',
      ],
      backoffRate: 1,
      maxAttempts: 3,
      interval: retryInterval,
    });

    const customValidationFailedState = new tasks.EventBridgePutEvents(
      this,
      customValidationFailedStateName,
      {
        entries: [
          {
            detail: TaskInput.fromObject({
              Message: 'Validation Failed',
            }),
            detailType: 'ValidationFailed',
            source: 'LocalTestingSource',
          },
        ],
      }
    );

    const validationExceptionState = new tasks.EventBridgePutEvents(
      this,
      validationExceptionStateName,
      {
        entries: [
          {
            detail: TaskInput.fromObject({
              Message: 'Validation Exception',
            }),
            detailType: 'ValidationException',
            source: 'LocalTestingSource',
          },
        ],
      }
    );

    const validation = new sfn.Parallel(this, validationStateName, {
      resultSelector: {
        identityResult: JsonPath.stringAt('$[0].identity'),
        addressResult: JsonPath.stringAt('$[1].address'),
      },
      resultPath: JsonPath.stringAt('$.results'),
    });

    const detectSentimentState = new sfn.CustomState(
      this,
      detectSentimentStateName,
      {
        stateJson: {
          Type: 'Task',
          Resource: 'arn:aws:states:::aws-sdk:comprehend:detectSentiment',
          Parameters: {
            LanguageCode: 'en',
            'Text.$': JsonPath.stringAt('$.data.comments'),
          },
          ResultSelector: {
            'sentimentAnalysis.$': JsonPath.stringAt('$'),
          },
          ResultPath: JsonPath.stringAt('$.results'),
          Retry: [
            {
              ErrorEquals: ['InternalServerException'],
              IntervalSeconds: 1,
              MaxAttempts: 3,
              BackoffRate: 1,
            },
          ],
        },
      }
    );

    const negativeSentimentDetectedState = new tasks.EventBridgePutEvents(
      this,
      negativeSentimentDetectedStateName,
      {
        entries: [
          {
            detail: TaskInput.fromObject({
              Message: 'Negative Sentiment Detected',
              Data: JsonPath.stringAt('$.data'),
            }),
            detailType: 'NegativeSentiment',
            source: 'LocalTestingSource',
          },
        ],
      }
    );

    const addToFollowUpState = new tasks.DynamoPutItem(
      this,
      addToFollowUpStateName,
      {
        inputPath: '$.data',
        table: ddb.Table.fromTableName(this, 'FollowUpTable', 'FollowUpTable'),
        item: {
          PK: DynamoAttributeValue.fromString(
            JsonPath.stringAt('$.identity.email')
          ),
        },
        resultSelector: {
          dbUpdateStatusCode: JsonPath.stringAt(
            '$.SdkHttpMetadata.HttpStatusCode'
          ),
        },
        resultPath: JsonPath.stringAt('$.results'),
      }
    );

    const customerAddedToFollowupState = new tasks.EventBridgePutEvents(
      this,
      customerAddedToFollowupStateName,
      {
        inputPath: JsonPath.stringAt('$.data'),
        entries: [
          {
            detail: TaskInput.fromObject({
              Message: 'Customer Added for follow up',
              EmailAddress: JsonPath.stringAt('$.identity.email'),
            }),
            detailType: 'CustomerAdded',
            source: 'LocalTestingSource',
          },
        ],
      }
    );

    const isPositiveSentimentState = new sfn.Choice(
      this,
      isPositiveSentimentStateName
    );

    const definition = validation
      .branch(checkIdentityState, checkAddressState)
      .addCatch(customValidationFailedState, {
        errors: ['CustomValidationError'],
      })
      .addCatch(validationExceptionState, { errors: ['States.ALL'] })
      .next(detectSentimentState)
      .next(
        isPositiveSentimentState
          .when(
            sfn.Condition.stringEquals(
              JsonPath.stringAt('$.results.sentimentAnalysis.Sentiment'),
              'POSITIVE'
            ),
            addToFollowUpState.next(customerAddedToFollowupState)
          )
          .otherwise(negativeSentimentDetectedState)
      );

    const sm = new sfn.StateMachine(this, stateMachineName, {
      stateMachineName,
      definition,
    });
  }
}
