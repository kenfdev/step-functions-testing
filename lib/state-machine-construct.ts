import { Construct } from 'constructs';
import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import { TaskInput } from 'aws-cdk-lib/aws-stepfunctions';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { Duration } from 'aws-cdk-lib';

type StateMachineProps = {
  stateMachineName?: string;
  lambdaStateName?: string;
  sqsStateName?: string;
  retryInterval?: Duration;
};

export class StateMachineConstruct extends Construct {
  constructor(scope: Construct, id: string, props: StateMachineProps = {}) {
    super(scope, id);

    const {
      lambdaStateName = 'LambdaState',
      sqsStateName = 'SQSState',
      stateMachineName,
      retryInterval = Duration.seconds(2),
    } = props;

    const lambdaState = new tasks.LambdaInvoke(this, lambdaStateName, {
      lambdaFunction: lambda.Function.fromFunctionArn(
        this,
        'lambdaInvoke',
        'arn:aws:lambda:us-west-2:123456789012:function:my-function'
      ),
      payload: TaskInput.fromObject({
        'Payload.$': '$',
        FunctionName: 'HelloWorldFunction',
      }),
      retryOnServiceExceptions: false,
    }).addRetry({
      errors: ['States.ALL'],
      backoffRate: 2,
      maxAttempts: 3,
      interval: retryInterval,
    });

    const sqsState = new tasks.SqsSendMessage(this, sqsStateName, {
      queue: sqs.Queue.fromQueueArn(
        this,
        'sqs',
        'arn:aws:sqs:us-west-2:444455556666:queue1'
      ),
      messageBody: TaskInput.fromJsonPathAt('$'),
    });

    const definition = lambdaState.next(sqsState);

    new sfn.StateMachine(this, 'SampleStateMachine', {
      stateMachineName,
      definition,
    });
  }
}
