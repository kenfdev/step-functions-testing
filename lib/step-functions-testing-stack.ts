import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { StateMachineConstruct } from './state-machine-construct';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class StepFunctionsTestingStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    new StateMachineConstruct(this, 'StateMachine');

    // example resource
    // const queue = new sqs.Queue(this, 'StepFunctionsTestingQueue', {
    //   visibilityTimeout: cdk.Duration.seconds(300)
    // });
  }
}
