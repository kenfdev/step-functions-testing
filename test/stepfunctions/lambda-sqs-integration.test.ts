import {
  MockedResponse,
  StateMachineTestCase,
  StateMachineTestDefinition,
  StepFunctionsMockConfig,
} from 'stepfunctions-testing';

import * as cdk from 'aws-cdk-lib';
import { StateMachineConstruct } from '../../lib/state-machine-construct';
import { createJestTestFromMockConfig } from '../helpers/stepfunctions';
import { Duration } from 'aws-cdk-lib';
import { extractStateMachineAsls } from '../helpers/stepfunctions/cdk-to-asl';

// TODO: dummy code
type LambdaResponse = {
  StatusCode: number;
  Payload: {
    StatusCode: number;
    body: any;
  };
};

export const stateNames = {
  LambdaStateName: 'LambdaState',
  SqsStateName: 'SqsState',
} as const;

type StateNameKeys = keyof typeof stateNames;
type StateName = typeof stateNames[StateNameKeys];

export const stateMachineName = 'LambdaSQSIntegration';

const mockedLambdaSuccessResponse = new MockedResponse(
  'MockedLambdaSuccess'
).return<LambdaResponse>({
  StatusCode: 200,
  Payload: {
    StatusCode: 200,
    body: 'Hello from Lambda!',
  },
});

const lambdaMockedResourceNotReadyResponse = new MockedResponse(
  'LambdaMockedResourceNotReady'
).throw('Lambda.ResourceNotReadyException', 'Lambda resource is not ready.');

const mockedSqsSuccessResponse = new MockedResponse('MockedSQSSuccess').return({
  MD5OfMessageBody: '3bcb6e8e-7h85-4375-b0bc-1a59812c6e51',
  MessageId: '3bcb6e8e-8b51-4375-b0bc-1a59812c6e51',
});

const mockedLambdaRetryResponse = new MockedResponse('MockedLambdaRetry')
  .throw('Lambda.ResourceNotReadyException', 'Lambda resource is not ready.')
  .throw('Lambda.TimeoutException', 'Lambda timed out.', 1)
  .return<LambdaResponse>({
    StatusCode: 200,
    Payload: {
      StatusCode: 200,
      body: 'Hello from Lambda!',
    },
  });

const stateMachineTestDefinition = new StateMachineTestDefinition(
  stateMachineName
)
  .addTestCase(
    new StateMachineTestCase<StateName>('HappyPath')
      .addMockedState(stateNames.LambdaStateName, mockedLambdaSuccessResponse)
      .addMockedState(stateNames.SqsStateName, mockedSqsSuccessResponse)
  )
  .addTestCase(
    new StateMachineTestCase<StateName>('RetryPath')
      .addMockedState(stateNames.LambdaStateName, mockedLambdaRetryResponse)
      .addMockedState(stateNames.SqsStateName, mockedSqsSuccessResponse)
  );

const config = new StepFunctionsMockConfig();
config.addTestDefinition(stateMachineTestDefinition);

const stack = new cdk.Stack();
new StateMachineConstruct(stack, 'StateMachine', {
  lambdaStateName: stateNames.LambdaStateName,
  sqsStateName: stateNames.SqsStateName,
  stateMachineName: stateMachineName,
  retryInterval: Duration.seconds(0),
});
// extract the Asl part
const asls = extractStateMachineAsls(stack);

describe(
  'Lambda SQS integration',
  createJestTestFromMockConfig(config, asls[0])
);
