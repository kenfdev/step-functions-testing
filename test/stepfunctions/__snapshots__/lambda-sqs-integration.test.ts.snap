// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lambda SQS integration LambdaSQSIntegration HappyPath 1`] = `
Array [
  Object {
    "ExecutionStarted": Array [
      Object {
        "detail": Object {
          "input": "{}",
          "inputDetails": Object {
            "truncated": false,
          },
          "roleArn": "arn:aws:iam::123456789012:role/service-role/LambdaSQSIntegration",
        },
        "type": "ExecutionStarted",
      },
    ],
    "TaskStateEnteredLambdaState": Array [
      Object {
        "detail": Object {
          "input": "{}",
          "inputDetails": Object {
            "truncated": false,
          },
          "name": "LambdaState",
        },
        "type": "TaskStateEntered",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"FunctionName\\":\\"arn:aws:lambda:us-west-2:123456789012:function:my-function\\",\\"Payload\\":{\\"FunctionName\\":\\"HelloWorldFunction\\",\\"Payload\\":{}}}",
          "region": "us-east-1",
          "resource": "invoke",
          "resourceType": "lambda",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "output": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "outputDetails": Object {
            "truncated": false,
          },
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskSucceeded",
      },
      Object {
        "detail": Object {
          "name": "LambdaState",
          "output": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "TaskStateExited",
      },
      Object {
        "detail": Object {
          "input": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "inputDetails": Object {
            "truncated": false,
          },
          "name": "SqsState",
        },
        "type": "TaskStateEntered",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"QueueUrl\\":\\"https://sqs.us-west-2.amazonaws.com/444455556666/queue1\\",\\"MessageBody\\":{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}}",
          "region": "us-east-1",
          "resource": "sendMessage",
          "resourceType": "sqs",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "sendMessage",
          "resourceType": "sqs",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
          "resource": "sendMessage",
          "resourceType": "sqs",
        },
        "type": "TaskSucceeded",
      },
      Object {
        "detail": Object {
          "name": "SqsState",
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "TaskStateExited",
      },
      Object {
        "detail": Object {
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "ExecutionSucceeded",
      },
    ],
  },
]
`;

exports[`Lambda SQS integration LambdaSQSIntegration RetryPath 1`] = `
Array [
  Object {
    "ExecutionStarted": Array [
      Object {
        "detail": Object {
          "input": "{}",
          "inputDetails": Object {
            "truncated": false,
          },
          "roleArn": "arn:aws:iam::123456789012:role/service-role/LambdaSQSIntegration",
        },
        "type": "ExecutionStarted",
      },
    ],
    "TaskStateEnteredLambdaState": Array [
      Object {
        "detail": Object {
          "input": "{}",
          "inputDetails": Object {
            "truncated": false,
          },
          "name": "LambdaState",
        },
        "type": "TaskStateEntered",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"FunctionName\\":\\"arn:aws:lambda:us-west-2:123456789012:function:my-function\\",\\"Payload\\":{\\"FunctionName\\":\\"HelloWorldFunction\\",\\"Payload\\":{}}}",
          "region": "us-east-1",
          "resource": "invoke",
          "resourceType": "lambda",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "cause": "Lambda resource is not ready.",
          "error": "Lambda.ResourceNotReadyException",
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskFailed",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"FunctionName\\":\\"arn:aws:lambda:us-west-2:123456789012:function:my-function\\",\\"Payload\\":{\\"FunctionName\\":\\"HelloWorldFunction\\",\\"Payload\\":{}}}",
          "region": "us-east-1",
          "resource": "invoke",
          "resourceType": "lambda",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "cause": "Lambda timed out.",
          "error": "Lambda.TimeoutException",
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskFailed",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"FunctionName\\":\\"arn:aws:lambda:us-west-2:123456789012:function:my-function\\",\\"Payload\\":{\\"FunctionName\\":\\"HelloWorldFunction\\",\\"Payload\\":{}}}",
          "region": "us-east-1",
          "resource": "invoke",
          "resourceType": "lambda",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "cause": "Lambda timed out.",
          "error": "Lambda.TimeoutException",
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskFailed",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"FunctionName\\":\\"arn:aws:lambda:us-west-2:123456789012:function:my-function\\",\\"Payload\\":{\\"FunctionName\\":\\"HelloWorldFunction\\",\\"Payload\\":{}}}",
          "region": "us-east-1",
          "resource": "invoke",
          "resourceType": "lambda",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "output": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "outputDetails": Object {
            "truncated": false,
          },
          "resource": "invoke",
          "resourceType": "lambda",
        },
        "type": "TaskSucceeded",
      },
      Object {
        "detail": Object {
          "name": "LambdaState",
          "output": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "TaskStateExited",
      },
      Object {
        "detail": Object {
          "input": "{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}",
          "inputDetails": Object {
            "truncated": false,
          },
          "name": "SqsState",
        },
        "type": "TaskStateEntered",
      },
      Object {
        "detail": Object {
          "heartbeatInSeconds": undefined,
          "parameters": "{\\"QueueUrl\\":\\"https://sqs.us-west-2.amazonaws.com/444455556666/queue1\\",\\"MessageBody\\":{\\"StatusCode\\":200,\\"Payload\\":{\\"StatusCode\\":200,\\"body\\":\\"Hello from Lambda!\\"}}}",
          "region": "us-east-1",
          "resource": "sendMessage",
          "resourceType": "sqs",
          "timeoutInSeconds": undefined,
        },
        "type": "TaskScheduled",
      },
      Object {
        "detail": Object {
          "resource": "sendMessage",
          "resourceType": "sqs",
        },
        "type": "TaskStarted",
      },
      Object {
        "detail": Object {
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
          "resource": "sendMessage",
          "resourceType": "sqs",
        },
        "type": "TaskSucceeded",
      },
      Object {
        "detail": Object {
          "name": "SqsState",
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "TaskStateExited",
      },
      Object {
        "detail": Object {
          "output": "{\\"MD5OfMessageBody\\":\\"3bcb6e8e-7h85-4375-b0bc-1a59812c6e51\\",\\"MessageId\\":\\"3bcb6e8e-8b51-4375-b0bc-1a59812c6e51\\"}",
          "outputDetails": Object {
            "truncated": false,
          },
        },
        "type": "ExecutionSucceeded",
      },
    ],
  },
]
`;
